freqtrade/tests/test_persistence.py

```diff
@@ -21,77 +21,22 @@ def test_init_create_session(default_conf, mocker):
     assert 'Session' in type(Trade.session).__name__
 
 
-def test_init_dry_run_db(default_conf, mocker):
-    default_conf.update({'dry_run_db': True})
-    mocker.patch.dict('freqtrade.persistence._CONF', default_conf)
-
-    # First, protect the existing 'tradesv3.dry_run.sqlite' (Do not delete user data)
-    dry_run_db = 'tradesv3.dry_run.sqlite'
-    dry_run_db_swp = dry_run_db + '.swp'
+def test_init_custom_db_url(default_conf, mocker):
+    conf = deepcopy(default_conf)
 
-    if os.path.isfile(dry_run_db):
-        os.rename(dry_run_db, dry_run_db_swp)
+    # Update path to a value other than default, but still in-memory
+    conf.update({'db_url': 'sqlite:///'})
+    mocker.patch.dict('freqtrade.persistence._CONF', conf)
 
     # Check if the new tradesv3.dry_run.sqlite was created
-    init(default_conf)
-    assert os.path.isfile(dry_run_db) is True
-
-    # Delete the file made for this unitest and rollback to the previous
-    # tradesv3.dry_run.sqlite file
-
-    # 1. Delete file from the test
-    if os.path.isfile(dry_run_db):
-        os.remove(dry_run_db)
-
-    # 2. Rollback to the initial file
-    if os.path.isfile(dry_run_db_swp):
-        os.rename(dry_run_db_swp, dry_run_db)
-
-
-def test_init_dry_run_without_db(default_conf, mocker):
-    default_conf.update({'dry_run_db': False})
-    mocker.patch.dict('freqtrade.persistence._CONF', default_conf)
-
-    # First, protect the existing 'tradesv3.dry_run.sqlite' (Do not delete user data)
-    dry_run_db = 'tradesv3.dry_run.sqlite'
-    dry_run_db_swp = dry_run_db + '.swp'
-
-    if os.path.isfile(dry_run_db):
-        os.rename(dry_run_db, dry_run_db_swp)
-
-    # Check if the new tradesv3.dry_run.sqlite was created
-    init(default_conf)
-    assert os.path.isfile(dry_run_db) is False
-
-    # Rollback to the initial 'tradesv3.dry_run.sqlite' file
-    if os.path.isfile(dry_run_db_swp):
-        os.rename(dry_run_db_swp, dry_run_db)
+    init(conf)
 
 
 def test_init_prod_db(default_conf, mocker):
     default_conf.update({'dry_run': False})
     mocker.patch.dict('freqtrade.persistence._CONF', default_conf)
 
-    # First, protect the existing 'tradesv3.sqlite' (Do not delete user data)
-    prod_db = 'tradesv3.sqlite'
-    prod_db_swp = prod_db + '.swp'
-
-    if os.path.isfile(prod_db):
-        os.rename(prod_db, prod_db_swp)
-
-    # Check if the new tradesv3.sqlite was created
     init(default_conf)
-    assert os.path.isfile(prod_db) is True
```

Comment:
Not sure about removing all of these tests ... 
i think we should  still test if the file is created successfully - even if the path is not hardcoded anymore?

Reply:
Agreed, I removed them because I think the tests shouldn't do any IO whether network or fs. I will check if I can mock this away somehow. Already tried to create a db within `/tmp` with `tempfile.mkdtemp`, but I have the feeling that SQLAlchemy doesn't allow this if `/tmp` is a tmpfs